src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -851,6 +851,22 @@ public INamedTypeSymbol GetTypeByMetadataName(string fullyQualifiedMetadataName)
         /// </summary>
         public abstract ImmutableArray<Diagnostic> GetDiagnostics(CancellationToken cancellationToken = default(CancellationToken));
 
+        internal void EnsureCompilationCompleted()
```

Comment:
The name should make it clear that this is about event queue. Right now the name is confusing.